android/app/src/main/java/au/com/robin/sms/service/SubscriberService.kt

```diff
@@ -38,6 +39,8 @@ private const val NOTIFICATION_GROUP_ID = "com.robin.sms.NOTIFICATION_GROUP_SERV
  * - https://robertohuertas.com/2019/06/29/android_foreground_services/
  */
 class SubscriberService : Service() {
+    private var repository: Repository? = null
```

Comment:
Provided that `wsConnection` can be instantiated in `onStartCommand`  (see comment below), I think you can make `repository` unnullable as well by instantiating also in `onStartCommand`.